scripts/get_gnomad_lof_variants.py

```diff
@@ -111,22 +119,53 @@ def load_gnomad_v3_variants():
     return ds
 
 
-def add_liftover_chain_to_reference(reference_genome):
+def add_liftover_mapping(ds, reference_genome):
+    """
+    Load the gnomAD liftover Hail tables depending on the reference genome
+    Add a mapping from the hg37 sites to the hg38 sites to the dataset (or vice versa)
+    """
     if reference_genome == "GRCh37":
-        hl.get_reference(reference_genome).add_liftover(
-            "gs://hail-common/references/grch37_to_grch38.over.chain.gz",
-            "GRCh38",
+        exome_liftover_mapping = hl.read_table(GNOMAD_V2_EXOME_LIFTOVER)
+        genome_liftover_mapping = hl.read_table(GNOMAD_V2_GENOME_LIFTOVER)
+
+        # hg38 fields: ['locus', 'alleles']
+        ds_exome = ds.annotate(
+            liftover_locus=exome_liftover_mapping[ds.locus, ds.alleles].locus,
+            liftover_alleles=exome_liftover_mapping[ds.locus, ds.alleles].alleles,
         )
+        ds_genome = ds.annotate(
+            liftover_locus=genome_liftover_mapping[ds.locus, ds.alleles].locus,
+            liftover_alleles=genome_liftover_mapping[ds.locus, ds.alleles].alleles,
+        )
+
     elif reference_genome == "GRCh38":
-        hl.get_reference(reference_genome).add_liftover(
-            "gs://hail-common/references/grch38_to_grch37.over.chain.gz",
-            "GRCh37",
+        exome_liftover_mapping = hl.read_table(GNOMAD_V3_EXOME_LIFTOVER)
+        genome_liftover_mapping = hl.read_table(GNOMAD_V3_GENOME_LIFTOVER)
```

Comment:
Are these tables keyed by `['original_locus', 'original_alleles']`? If so I think you may need to re-key these tables so that `['locus', 'alleles']` are the new keys before annotating if going backwards.

Reply:
There are two different sets of liftover tables (V2 or V3 liftover) and each set is keyed by either the hg37 locus and alleles or the hg38 locus and alleles.

The V2 liftover tables which go from hg37 to hg38 are keyed by the hg37 locus and alleles `['original_locus', 'original_alleles']`, so the liftover mapping extracts the hg38 fields `['locus', 'alleles']`. 

The V3 liftover tables from hg38 to hg37 are keyed by the hg38 locus and alleles `['locus', 'alleles']`, so the liftover mapping extracts the hg37 fields `['original_locus', 'original_alleles']`. Hope this makes sense

Reply:
Oh, right! Makes perfect sense...I just can't read code properly 😆

Reply:
All good, its genuinely quite confusing in the code here. If you actually get to look at the Hail tables, its a lot clearer since the `locus` and `original_locus` fields have the type underneath the column headers, which instantly clarifies which one is hg38 and which is hg37.